kyuubi-main/src/test/scala/org/apache/kyuubi/engine/spark/InitializeSQLSuite.scala

```diff
@@ -40,12 +45,31 @@ class InitializeSQLSuite extends WithKyuubiServer with JDBCTestUtils {
 
   test("KYUUBI-457: Support configurable initialize sql statement for engine startup") {
     withJdbcStatement() { statement =>
-      val result = statement.executeQuery("SELECT * FROM INIT_DB.test")
+      val result = statement.executeQuery("SELECT * FROM INIT_DB.test WHERE a = 1")
       assert(result.next())
       assert(result.getInt(1) == 1)
       assert(!result.next())
     }
   }
 
+  test("Support configurable initialize sql statement for engine session creation") {
+    var currentSessionCnt: Long = -1
+    withJdbcStatement() { statement =>
+      val result = statement.executeQuery("SELECT COUNT(*) FROM INIT_DB.test WHERE a = 2")
+      assert(result.next())
+      currentSessionCnt = result.getLong(1)
+      assert(currentSessionCnt >= 1)
+      assert(!result.next())
+    }
+    // new session
+    withJdbcStatement() { statement =>
+      val result = statement.executeQuery("SELECT COUNT(*) FROM INIT_DB.test WHERE a = 2")
+      assert(result.next())
+      // use great than or equals to support concurrent test
+      assert(result.getLong(1) >= currentSessionCnt + 1)
+      assert(!result.next())
+    }
```

Comment:
should we add a test for single spark session mode?

Reply:
yes, will update

Reply:
updated, please review again